improver/nbhood/use_nbhood.py

```diff
@@ -402,9 +403,10 @@ def process(self, cube):
         # Loop over any extra dimensions
         cubelist = iris.cube.CubeList([])
         for slice_3d in cube.slices([self.coord_masked, yname, xname]):
-            collapsed_slice = slice_3d.collapsed(
-                self.coord_masked, iris.analysis.MEAN, weights=weights)
+            collapsed_slice = collapsed(slice_3d, self.coord_masked,
```

Comment:
I notice that the CollapseMaskedNeighbourhoodCoordinate plugin contains a method called `remove_collapsed_coord_refs`. I think that following your creation of a `collapsed` function, the call to the `remove_collapsed_coord_refs` method could be replaced by just `result.remove_coord(self.coord_masked)`, as the cell method operations in `remove_collapsed_coord_refs` are now redundant following your creation of the `collapsed` function.

Reply:
The `remove_collapsed_coord_refs` method can be removed without breaking any tests, just one line of code needs to be moved out of the loop.